src/validator.ts

```diff
@@ -0,0 +1,43 @@
+import Ajv, { Schema } from 'ajv';
+import addFormats from 'ajv-formats';
+import lodash from 'lodash';
+import { ValidationError, betterAjvErrors } from '@apideck/better-ajv-errors';
+import { createDebug } from './utils/debug';
+
+const debug = createDebug('validator');
+
+export const ajvConfigValidator = addFormats(
+  new Ajv({
+    useDefaults: true,
+    allErrors: true,
+    verbose: true,
+    keywords: ['x-env-value'],
+  }),
+  ['date-time', 'time', 'date', 'email', 'hostname', 'ipv4', 'ipv6', 'uri', 'uuid', 'regex', 'uri-template']
+);
+
+export const ajvOptionsValidator = new Ajv({
+  useDefaults: true,
+  coerceTypes: true,
+  allErrors: true,
+  verbose: true,
+});
+
+export function validate<T>(ajv: Ajv, schema: Schema, data: unknown): [ValidationError[], undefined] | [undefined, T] {
```

Comment:
It seems that `schema` can't be a boolean. Consider using `SchemaObject` instead of `Schema`
```suggestion
export function validate<T>(ajv: Ajv, schema: SchemaObject, data: unknown): [ValidationError[], undefined] | [undefined, T] {
```

It will also affects the validator.spec.ts file (no need to test the schema as boolean)

Reply:
fixed